patternfly/patternfly-elements
# feat: Update pfe-content-set to accept custom container queries

## Update pfe-content-set to accept custom container queries

Discovery doc:
* https://docs.google.com/document/d/15sfSRWp4OAy2ndbXqHp-nuPjj7kYbyP1zbTqxiHfvo8/edit#heading=h.ughne74p2l4j

Sandbox link:
* https://web-dev-drupal-alazzara.dev.a1.vary.redhat.com/en/pfe-content-set-test

---

### What has changed and why
- Updated pfe-content-set.js - to update offsetWidth based on a value given in a new data attribute called "pfe-breakpoint". This was done to fix and issue with tabs spilling off the page by giving the ability to set when the tabs change to accordions. You can input values with or without px and it will work (the code strips out any non-numeric values). This is to insure the code wont break if someone happens to put 500px instead of just 500.


### Testing instructions
Test cases:
1. If I set the value of the attribute pfe-breakpoint to 500
    a. And I reload the page at window size 300, i expect that
        i. The content set should become accordions
    b. And I reload the page at window size 550, i expect that
        i. The content set should become tabs
2. If I set the value of the attribute pfe-breakpoint to 900
    a. And I reload the page at window size 800, i expect that
        i. The content set should become accordions
    b. And I reload the page at window size 1000, i expect that
        i. The content set should become tabs
3. If I  do not set the value of the attribute pfe-breakpoint 
    a. And I reload the page at window size 600, i expect that
        i. The content set should become accordions
    b. And I reload the page at window size 800, i expect that
        i. The content set should become tabs
4. If I load the page at window size 600, And then I use web inspector to add the attribute pfe-breakpoint 
    a. with a value of 400 after the page has loaded
        i. The content set should become accordions
    b. with a value of 650 after the page has loaded
        i. The content set should become tabs

Edge case:
    1. Test using px and no px in the value of pfe-breakpoint


#### Browser requirements
Your component should work in all of the following environments:



### Ready-for-merge Checklist



**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
